src/Okta.DNX.OAuth.ResourceServer/Models/TodoRepository.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace Okta.DNX.OAuth.ResourceServer.Models
+{
+    public class TodoRepository : ITodoRepository
+    {
+        private static ConcurrentDictionary<string, TodoItem> _todos =
+              new ConcurrentDictionary<string, TodoItem>();
+
+        public TodoRepository()
+        {
+            Add(new TodoItem { Name = "Item1" });
+        }
+
+        public IEnumerable<TodoItem> GetAll()
+        {
+            return _todos.Values;
+        }
+
+        public void Add(TodoItem item)
+        {
+            item.Key = Guid.NewGuid().ToString();
+            _todos[item.Key] = item;
+        }
+
+        public TodoItem Find(string key)
+        {
+            TodoItem item;
+            _todos.TryGetValue(key, out item);
+            return item;
+        }
+
+        public TodoItem Remove(string key)
+        {
+            TodoItem item;
+            _todos.TryRemove(key, out item);
+            return item;
+        }
+
+        public void Update(TodoItem item)
+        {
+            _todos[item.Key] = item;
```

Comment:
[TryUpdate](https://msdn.microsoft.com/en-us/library/dd287117(v=vs.110).aspx), and actually, you could Update method could return `bool`